tests/test_events_api.py

```diff
@@ -0,0 +1,172 @@
+        assert self.event_api._source == "XXX"
+
+    def test_set_resource(self) -> None:
+        # pylint: disable=protected-access
+        self.event_api.set_resources(["x", "Y"])
+        assert self.event_api._resources == ["x", "Y"]
+
+    def test_register(self) -> None:
+        event = MyTestEvent({"x": 1})
+        self.event_api.register(event)
+
+        assert self.event_api.get_all_pending_events() == [event]
+
+    def test_get_all_registered_events(self) -> None:
+        event_1 = MyTestEvent({"x": 1})
+        event_2 = MyTestEvent({"x": 2})
+        self.event_api.register(event_1)
+        self.event_api.register(event_2)
+        assert self.event_api.get_all_pending_events() == [event_1, event_2]
+
+    @patch.object(Boto3Client, "eventbridge")
+    def test_send(self, mock_send: MagicMock) -> None:
+        event = MyTestEvent({"x": 1})
+        self.event_api.register(event)
+
+        self.event_api.send()
+
+        mock_send.put_events.assert_called_once_with(
+            Entries=[
+                {
+                    "Detail": '{"x": 1}',
+                    "DetailType": "MY_TEST_EVENT",
+                    "EventBusName": "million-dollar-lambda-event-bus",
+                    "Resources": [],
+                    "Source": "million-dollar-lambda",
+                }
+            ]
+        )
+        assert self.event_api.get_all_sent_events() == [event]
+
+    def test__mark_sent_moves_events_from_pending_to_sent(self) -> None:
+        event = MyTestEvent({"x": 1})
+        self.event_api.register(event)
+
+        self.event_api._mark_sent()  # pylint: disable=protected-access
+
+        assert self.event_api.get_all_pending_events() == []
+        assert self.event_api.get_all_sent_events() == [event]
+        assert self.event_api.get_all_failed_events() == []
+
+    def test__mark_failed_moves_events_from_pending_to_failed(self) -> None:
+        event = MyTestEvent({"x": 1})
+        self.event_api.register(event)
+
+        self.event_api._mark_failed()  # pylint: disable=protected-access
+
+        assert self.event_api.get_all_pending_events() == []
+        assert self.event_api.get_all_sent_events() == []
+        assert self.event_api.get_all_failed_events() == [event]
+
+    def test_get_all_sent_events(self) -> None:
+        event = MyTestEvent({"x": 1})
+        self.event_api.register(event)
+
+        self.event_api._mark_sent()  # pylint: disable=protected-access
+
+        assert self.event_api.get_all_sent_events() == [event]
+
+    def test_get_all_failed_events(self) -> None:
+        event = MyTestEvent({"x": 1})
+        self.event_api.register(event)
+
+        self.event_api._mark_failed()  # pylint: disable=protected-access
+
+        assert self.event_api.get_all_failed_events() == [event]
+
+    @patch.object(Boto3Client, "eventbridge")
+    def test_send_no_events(self, mock_send: MagicMock) -> None:
+        self.event_api.send()
+
+        mock_send.assert_not_called()
```

Comment:
This mock will never be called no matter what happens. You should check it this way:
```python
mock_send.put_events.assert_not_called()
```